src/sinks/aws_cloudwatch_metrics.rs

```diff
@@ -177,38 +184,80 @@ impl RetryLogic for CloudWatchMetricsRetryLogic {
         match error {
             PutMetricDataError::HttpDispatch(_) => true,
             PutMetricDataError::InternalServiceFault(_) => true,
-            PutMetricDataError::Unknown(res) if res.status.is_server_error() => true,
+            PutMetricDataError::Unknown(res)
+                if res.status.is_server_error()
+                    || res.status == http::StatusCode::TOO_MANY_REQUESTS =>
+            {
+                true
+            }
             _ => false,
         }
     }
 }
 
-fn encode_events(events: Vec<Event>, namespace: String) -> Result<PutMetricDataInput, ()> {
+fn timestamp_to_string(timestamp: DateTime<Utc>) -> String {
+    timestamp.to_rfc3339_opts(SecondsFormat::Millis, true)
+}
+
+fn encode_events(
+    state: &mut CloudWatchMetricsState,
+    events: Vec<Event>,
+    namespace: String,
+) -> Result<PutMetricDataInput, ()> {
```

Comment:
does an option make more sense here than a result since there is no error type?

Reply:
You are right. Seems like even plain `PutMetricDataInput` will do.